test/stablex/batch_exchange.js

```diff
@@ -1654,6 +1654,44 @@ contract("BatchExchange", async accounts => {
       }
     })
   })
+  describe("getEncodedUserOrdersPaginated()", async () => {
+    it("returns correct orders considering offset and pageSize", async () => {
+      const batchExchange = await setupGenericStableX()
+      const zeroBN = new BN(0)
+      const tenBN = new BN(10)
+      const twentyBN = new BN(20)
+
+      const batchId = (await batchExchange.getCurrentBatchId.call()).toNumber()
+      const OrderInfo = {
+        user: user_1.toLowerCase(),
+        sellTokenBalance: zeroBN,
+        buyToken: 1,
+        sellToken: 0,
+        validFrom: batchId,
+        validUntil: batchId + 10,
+        priceNumerator: twentyBN,
+        priceDenominator: tenBN,
+        remainingAmount: tenBN,
+      }
+
+      // Place 3 orders
+      for (let i = 0; i < 3; i++) {
+        await batchExchange.placeOrder(
+          OrderInfo.buyToken,
+          OrderInfo.sellToken,
+          OrderInfo.validUntil,
+          OrderInfo.priceNumerator.addn(i),
+          OrderInfo.priceDenominator
+        )
+      }
+
+      // get 2nd order with getEncodedUserOrdersPaginated(user_1, 1, 1)
+      const expectedOrder = OrderInfo
+      expectedOrder.priceNumerator = expectedOrder.priceNumerator.addn(1)
```

Comment:
ubertnit (and more high-level feedback): I find this unit test a bit logic intense. I'd just create orders with all value 0, all values 1, 2, and then assert that we get the one 1 all values 1.

Reply:
Yes... It kinda seems the readers interpretation is obfuscated by the definition of the orders more than the point itself. I wonder how we could make this cleaner? Maybe a `deepEquals`  on the order (that shouldn't refer to the values itself, but more the order... Shouldn't comparison of the `orderId` be a reasonable way to distinguish orders?